### PR TITLE
RM-7401: Edge loads test site pages slower than other browsers

### PR DESCRIPTION
**Pull request by iam-mholle**
https://re-motion.atlassian.net/browse/RM-7401

### CODE REVIEW

File: Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericPageTestCaseFactoryBase`.cs [hunk 1]
```diff
@@ -22,13 +22,16 @@
...
 using Remotion.Utilities;
+using Remotion.Web.Development.WebTesting.Utilities;
 
 namespace Remotion.Web.Development.WebTesting.IntegrationTests.Infrastructure
 {
   /// <inheritdoc />
   public abstract class GenericTestPageTestCaseFactoryBase<TParameter> : GenericPageTestCaseFactoryBase
       where TParameter : IGenericTestPageParameter, new()
   {
+    private static Lazy<Uri> _webApplicationRoot;
```
Comment (Stefan-Ilic) [inline Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericPageTestCaseFactoryBase`.cs:33]: It's static, rename to `s_webApplicationRoot`

Comment (Stefan-Ilic) [reply Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericPageTestCaseFactoryBase`.cs:33]: Also, Resharper is complaining about a "static field in a generic type"

File: Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericPageTestCaseFactoryBase`.cs [hunk 2]
```diff
@@ -57,8 +60,13 @@ protected void PrepareTest ([NotNull] GenericPageTestMethodAttribute attribute,
       ArgumentUtility.CheckNotNull ("helper", helper);
       ArgumentUtility.CheckNotNullOrEmpty ("control", control);
 
+      if (_webApplicationRoot == null)
+      {
+        _webApplicationRoot = new Lazy<Uri> (() => HostnameResolveHelper.ResolveHostname (new Uri (helper.TestInfrastructureConfiguration.WebApplicationRoot)));
```
Comment (Stefan-Ilic) [inline Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericPageTestCaseFactoryBase`.cs:65]: Should we leave a comment explaining why we're resolving it manually here as well?

File: Remotion/Web/Development.WebTesting.IntegrationTests/IntegrationTest.cs [hunk 3]
```diff
@@ -45,6 +47,7 @@ public void IntegrationTestOneTimeSetUp ()
     {
       _webTestHelper = WebTestHelper.CreateFromConfiguration<CustomWebTestConfigurationFactory>();
       _webTestHelper.OnFixtureSetUp (MaximizeMainBrowserSession);
+      s_webApplicationRoot = new Lazy<Uri> (() => HostnameResolveHelper.ResolveHostname (new Uri (_webTestHelper.TestInfrastructureConfiguration.WebApplicationRoot)));
```
Comment (Stefan-Ilic) [inline Remotion/Web/Development.WebTesting.IntegrationTests/IntegrationTest.cs:50]: Same question like in other `IntegrationTest.cs`

File: Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/HostnameResolveHelper.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+//
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using Remotion.Web.Development.WebTesting.Utilities;
+
+namespace Remotion.Web.Development.WebTesting.IntegrationTests.Infrastructure
+{
+  public static class HostnameResolveHelper
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/HostnameResolveHelper.cs:28]: officially, every type should have a summary when it's part of non-test code. Here, in the test infrastructure, it's a matter of "is it needed".

File: Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/IntegrationTest.cs [hunk 5]
```diff
@@ -47,6 +49,17 @@ public void IntegrationTestOneTimeSetUp ()
       _webTestHelper = WebTestHelper.CreateFromConfiguration<CustomWebTestConfigurationFactory>();
 
       _webTestHelper.OnFixtureSetUp (MaximizeMainBrowserSession);
+      s_webApplicationRoot = new Lazy<Uri> (
+          () =>
+          {
+            var uri = new Uri (_webTestHelper.TestInfrastructureConfiguration.WebApplicationRoot);
+
+            // RM-7401: Edge loads pages slower due to repeated hostname resolution.
+            if (_webTestHelper.BrowserConfiguration.IsEdge())
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.Development.WebTesting.IntegrationTests/IntegrationTest.cs:58]: do we already have a follow-up task to investigate/resolve the underlying issue? To be scheduled for somewhere in 2-3 months unless there's an earlier slot available.

**Comment by iam-mholle** [comment]
Added to master with b9f959e91d9cae0187ff8bfdd1061585defab625.

**Pull request closed without merging**